Source/Icebreaker/Bot/IcebreakerBot.cs

```diff
@@ -387,59 +594,81 @@ private async Task SendUnrecognizedInputMessageAsync(ITurnContext turnContext, A
         /// </summary>
         /// <param name="serviceUrl">The service url</param>
         /// <param name="teamId">The team id</param>
-        /// <param name="tenantId">The tenant id</param>
+        /// <param name="turnContext">Turn context</param>
         /// <param name="botInstaller">Person that has added the bot to the team</param>
         /// <returns>Tracking task</returns>
-        private Task SaveAddedToTeam(string serviceUrl, string teamId, string tenantId, string botInstaller)
+        private async Task SaveAddedToTeamAsync(string serviceUrl, string teamId, ITurnContext turnContext, string botInstaller)
         {
+            var tenantId = turnContext.Activity.GetChannelData<TeamsChannelData>().Tenant.Id;
+
             var teamInstallInfo = new TeamInstallInfo
             {
                 ServiceUrl = serviceUrl,
                 TeamId = teamId,
                 TenantId = tenantId,
                 InstallerName = botInstaller
             };
-            return this.dataProvider.UpdateTeamInstallStatusAsync(teamInstallInfo, true);
+
+            // add users in team
+            var botAdapter = turnContext.Adapter;
+            var members = await this.conversationHelper.GetTeamMembers(botAdapter, teamInstallInfo);
+            var userIds = new HashSet<string>();
+
+            foreach (var member in members)
+            {
+                var userId = this.GetChannelUserObjectId(member);
+                userIds.Add(userId);
+                await this.dataProvider.AddUserTeamAsync(tenantId, userId, teamId, serviceUrl);
+            }
+
+            teamInstallInfo.UserIds = userIds;
+            await this.dataProvider.UpdateTeamInstallStatusAsync(teamInstallInfo, true);
```

Comment:
It is worth noting this can cause problems with large teams. [CosmosDB has a single-item limit of 2MB](https://docs.microsoft.com/en-us/azure/cosmos-db/concepts-limits#per-item-limits), if these are guids stored as utf8 strings that is about 288 bytes per user. It means that a team can have at most ~7k users before this starts hitting size limit issues.

Reply:
Why do we need to store the user IDs in the team install info?

Reply:
When Icebreaker is removed from a team, we need to remove the team from the team members' `OptedIn` dictionary. Icebreaker can't access team information after it has been removed unless it has been previously stored.